packages/@aws-cdk/aws-s3/lib/pipeline-action.ts

```diff
@@ -53,12 +63,32 @@ export class PipelineSourceAction extends codepipeline.SourceAction {
       configuration: {
         S3Bucket: props.bucket.bucketName,
         S3ObjectKey: props.bucketKey,
-        PollForSourceChanges: props.pollForSourceChanges || true
+        PollForSourceChanges: shouldPollForSourceChanges(props),
       },
       ...props,
     });
 
+    if (props.pollForSourceChanges === false && props.cloudTrail === undefined) {
+      throw new Error('When not polling for source changes, ' +
```

Comment:
I'm actually having second thoughts about this validation :/. While the intention was good I think, I'm not sure if it's not too strict. For example, I might be using an `import`ed Bucket that already belongs to some Trail, and in that case I can safely say `pollForSourceChanges: false` and `cloudTrail: undefined`.

Thoughts on this?

Reply:
I like the validation, but there definitely is some interest in skipping for an imported bucket... Maybe you could attach a warning metadata entry to the resource instead of completely failing, and at a later stage, if/when we're able to detect "imported" entities, then you could switch it around and start failing again?